evaluations/subcoin_2_piewol.md

```diff
@@ -0,0 +1,191 @@
+# Evaluation
+
+- **Status:** in progress
+- **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/Subcoin.md
+- **Milestone:** 2
+
+| Number | Deliverable    | Accepted | Link | Notes |
+| -- | --  | ---    | --- | --- |
+| 0a.    | License                 | <ul><li>[x] </li></ul> | https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/LICENSE |  --  |
+| 0b.    | Documentation           | <ul><li>[ ] </li></ul> | Each module has its own docs. The rendered inline rustdoc is deployed at https://subcoin-project.github.io/subcoin/.  | some elements e.g. structs are still without inline doc |
+| 0c.    | Testing and Testing Guide | <ul><li>[x] </li></ul> |  https://github.com/subcoin-project/subcoin/tree/subcoin-milestone-2?tab=readme-ov-file#run-tests |  see general notes  |
+| 0d.    | Docker                  | <ul><li>[ ] </li></ul> | https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/Dockerfile The docker image is available at https://github.com/subcoin-project/subcoin/pkgs/container/subcoin/249545041?tag=v0.2.0  |  builds. see notes. |
+| 1.     | Block Verification	       | <ul><li>[ ] </li></ul> | 	https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/crates/sc-consensus-nakamoto/src/verification.rs |  how comes that there are no tests? Is this directly from ``rust-bitcoin``? |
```

Comment:
Actually, there are two tests in the verification module :P, see https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/crates/sc-consensus-nakamoto/src/verification/header_verify.rs#L266 and https://github.com/subcoin-project/subcoin/blob/subcoin-milestone-2/crates/sc-consensus-nakamoto/src/verification.rs#L569. I have to admit that the existing unit tests are not adequate as I haven't figured it out how to set up the entire test framework properly. The feature tests are now primarily conducted by running the node to sync the Bitcoin network locally. If the syncing is fine, it's probably fine (The existing two tests are spotted by the local syncing test). More comprehensive tests will be explored later.

Reply:
Sounds good. Keen to see how more testing will be added later.